src/main/java/com/google/devtools/build/lib/packages/semantics/BuildLanguageOptions.java

```diff
@@ -663,17 +663,6 @@ public final class BuildLanguageOptions extends OptionsBase {
       help = "If set to true, the ObjcProvider's APIs for linking info will be removed.")
   public boolean incompatibleObjcProviderRemoveLinkingInfo;
 
-  @Option(
-      name = "incompatible_disable_objc_library_transition",
-      defaultValue = "true",
-      documentationCategory = OptionDocumentationCategory.STARLARK_SEMANTICS,
-      effectTags = {OptionEffectTag.BUILD_FILE_SEMANTICS},
-      metadataTags = {OptionMetadataTag.INCOMPATIBLE_CHANGE},
-      help =
-          "Disable objc_library's custom transition and inherit "
-              + "from the top level target instead")
-  public boolean incompatibleDisableObjcLibraryTransition;
-
```

Comment:
This should probably be moved to the graveyard and made no-op to make it easier for people to test multiple Bazel versions. Can then be removed in Bazel 9.